reproduce-figures/prepare-scpca-portal-data.R

```diff
@@ -0,0 +1,309 @@
+output_dir_celltype_results <- file.path(output_dir_s3_files, "celltype_results")
+
+# Create directories which will be assumed to exist later
+fs::dir_create(output_dir_analysis_data)
+fs::dir_create(output_dir_celltype_results)
+
+# Define marker genes for recreating expression matrices used for consensus cell type dot plots
+marker_genes <- readr::read_tsv(marker_gene_ref_file) |>
+  dplyr::pull(ensembl_gene_id) |>
+  unique()
+
+# These are all output directories at the top-level of s3_files
+### Some need just processed, and some also need (un)filtered, but scripts specify which one
+### We can therefore retain all RDS files when copying; extra files will not be used either way
+standalone_sample_dirs <- c(
+  "SCPCS000001", 
+  "SCPCS000216", 
+  "SCPCS000251", 
+  "SCPCS000264"
+)
+
+# The samples' processed.rds files should be saved to output_dir_celltype_results
+celltype_results_samples <- c(
+  "SCPCS000001",
+  "SCPCS000002", 
+  "SCPCS000004", 
+  "SCPCS000252", 
+  "SCPCS000258",
+  "SCPCS000262", 
+  "SCPCS000222", 
+  "SCPCS000223", 
+  "SCPCS000224"
+)
+
+# These are the directories to save to output_dir_analysis_data
+# They should contain processed files organized as expected _AND_ with the bulk quant TSV file
+bulk_projects <- c(
+  "SCPCP000001", 
+  "SCPCP000002", 
+  "SCPCP000006", 
+  "SCPCP000009", 
+  "SCPCP000017"
+)
+
+
+# Define input files ------------------------
+
+# Read whitelist so we can check that all projects are present
+expected_projects <- readLines(opts$project_whitelist)
+
+# Define and check input projects
+input_zips <- list.files(
+  opts$portal_projects_dir, 
+  pattern = "^SCPCP\\d{6}_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_\\d+{4}-\\d+{2}-\\d+{2}\\.zip",
+  full.names = TRUE
+) |>
+  # name by project
+  purrr::set_names(
+    \(x) {stringr::str_split_i(basename(x), "_", 1)}
+  )
+stopifnot(
+  "The provided input directory does not contain all expected projects. Zip files for all projects listed in the `project-whitelist.txt` file should be present." = 
+    setequal(names(input_zips), expected_projects)
+)
+
+
+# Copy over the merged SCPCP000003 file ----------------
+output_merged_sce_path <- file.path(output_dir_s3_files, "SCPCP000003", "SCPCP000003_merged.rds")
+fs::dir_create(dirname(output_merged_sce_path))
+fs::file_copy(
+  opts$merged_sce_path, 
+  output_merged_sce_path
+)
+
+
+# Map over project zip files to organize files for reproducibility ------------------
+input_zips |>
+  purrr::iwalk(
+    \(project_zip, project_id){
+      
+      # temporary variables used for testing
+      #project_id <- "SCPCP000001"
+      #project_zip <- "/Users/sjspielman/Downloads/projects/SCPCP000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-04-03.zip"
+      
+      project_consensus_dir <- file.path(output_dir_consensus_files, project_id)
+      
+      # Unzip the directory, specifying the target in consensus
+      unzip(project_zip, exdir = project_consensus_dir)
```

Comment:
Okay after looking at this whole script, what I think I might do is create a scratch directory to hold the unzipped project, then copy over any processed SCE files/ files needed for bulk, and then read in the SCE file and output the TSVs to the consensus project directory. Then at the end you remove the scratch directory. I think that should simplify some of the code.